vllm/model_executor/layers/layernorm.py

```diff
@@ -158,8 +158,7 @@ def forward_cuda(
         x: torch.Tensor,
         residual: Optional[torch.Tensor] = None,
     ) -> Union[torch.Tensor, tuple[torch.Tensor, torch.Tensor]]:
-        if self.variance_size_override is not None:
-            return self.forward_native(x, residual)
+        return self.forward_native(x, residual)
```

Comment:
why was this change needed, the cuda rms norm kernels already read batch-invariant to me?

Reply:
> it is not

is it because it uses vectorize instructions? I think the kernel still processes one row per block.

Reply:
should've linked agains a specific commit, [here](https://github.com/vllm-project/vllm/blob/824a3f403f8cbb5b7984d33ca4741f67928c1d61/csrc/layernorm_kernels.cu#L394) is the line. it means different # of threads perform the reduction (sum_i x_i^2) depending on the number of tokens in the batch, which means the values will associate differently.